### PR TITLE
Added the ability to List all recommendations

### LINKED ISSUE #16: Need ability to List all recommendations
**As a** manager  
 **I need**  to create a service to get resource information from Database
 **So that** get the information regarding resource  
   
 ### Details and Assumptions
 * We put the information in a Structured database and pull information from that
   
 ### Acceptance Criteria  
   
 ```gherkin
 Given to get information from DB for a resource
 When a API call is made
 Then the API returns information regarding the resource
 ```

### PR DESCRIPTION
**Pull request by mkcz**
add list function and some tests

### CODE REVIEW

File: tests/test_routes.py [hunk 1]
```diff
@@ -124,3 +124,18 @@ def test_create_recommendation_no_content_type(self):
...
         self.assertEqual(resp.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
+    
+    def test_get_products_list(self):
+        """Get a list of items"""
+        self._create_recommendations(5)
+        resp = self.app.get(BASE_URL)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        data = resp.get_json()
+        self.assertEqual(len(data), 5)
+    # def test_get_products_list(self):
```
Comment (changtk07) [inline tests/test_routes.py]: Remove commented codes

File: tests/test_models.py [hunk 2]
```diff
@@ -137,6 +137,14 @@ def test_add_a_recommendation(self):
         self.assertEqual(recommendation.id, 1)
         recommendations = Recommendation.all()
         self.assertEqual(len(recommendations), 1)
+    def test_find_all(self):
```
Comment (changtk07) [inline tests/test_models.py:141]: Add a blank line before the function

File: service/routes.py [hunk 3]
```diff
@@ -31,7 +31,26 @@ def index():
 
 @app.route("/recommendations", methods=["GET"])
 def list_recommendations():
-    pass
+    #filter for future usage
```
Comment (changtk07) [inline service/routes.py]: Remove commented codes

File: service/models.py [hunk 4]
```diff
@@ -95,7 +95,11 @@ def deserialize(self, data: dict):
                 "Invalid recommendation: body of request contained bad or no data " + str(error)
             )
         return self
-
+    @classmethod
```
Comment (changtk07) [inline service/models.py:99]: Add a blank line before and after the function

**Review by changtk07 — CHANGES_REQUESTED** [review]
Need minor changes

File: service/routes.py [hunk 5]
```diff
@@ -31,7 +31,23 @@ def index():
 
 @app.route("/recommendations", methods=["GET"])
 def list_recommendations():
-    pass
+    products = Recommendation.all()
```
Comment (changtk07) [inline service/routes.py]: `products` -> `recommendations`

File: service/routes.py [hunk 6]
```diff
@@ -31,7 +31,23 @@ def index():
...
+    results = [product.serialize() for product in products]
+    app.logger.info("Returning %d products", len(results))
+    return make_response(jsonify(results), status.HTTP_200_OK)
+    """Returns all of the recommendation"""
```
Comment (changtk07) [inline service/routes.py:34]: Remove all codes after the return statement, include commented codes

**Review by changtk07 — CHANGES_REQUESTED** [review]
More minor changes needed

File: service/routes.py (hunk 5 above)
Comment (mkcz) [reply service/routes.py]: I think this is value name, what we get will be products

File: service/routes.py (hunk 6 above)
Comment (mkcz) [reply service/routes.py:34]: it is left for next spr

File: service/routes.py (hunk 5 above)
Comment (changtk07) [reply service/routes.py]: In the database we have recommendations, not products.
Out Recommendation model have attributes:
`id` : recommendation id
`src_product_id` :  source product id
`rec_product_id` : recommened product id
`type`: recommendation type

Comment (mkcz) [reply service/routes.py]: ok, but they still products, anyway, i will change it

File: service/routes.py (hunk 6 above)
Comment (changtk07) [reply service/routes.py:34]: Yeah keep it in your local, not remote. We need to follow PEP standard.

**Review by changtk07 — APPROVED** [review]

File: service/routes.py (hunk 6 above)
Comment (mkcz) [reply service/routes.py:34]: i will remove that, but you know left it in local is very dangers.

next time pull command will have a conflict and may destroy that

**Review by aianurag09 — APPROVED** [review]

**Pull request merged**